disnake/channel.py

```diff
@@ -448,13 +449,83 @@ async def edit(
             # the payload will always be the proper channel payload
             return self.__class__(state=self._state, guild=self.guild, data=payload)  # type: ignore
 
-    @utils.copy_doc(disnake.abc.GuildChannel.clone)
     async def clone(
-        self, *, name: Optional[str] = None, reason: Optional[str] = None
+        self,
+        *,
+        name: Optional[str] = None,
+        topic: Optional[str] = MISSING,
+        position: Optional[int] = MISSING,
+        nsfw: Optional[bool] = MISSING,
+        category: Optional[Snowflake] = MISSING,
+        slowmode_delay: Optional[int] = MISSING,
+        type: Optional[ChannelType] = MISSING,
+        default_auto_archive_duration: Optional[AnyThreadArchiveDuration] = MISSING,
+        flags: Optional[ChannelFlags] = MISSING,
```

Comment:
I'm not sure if having a `type` here is needed at all, it might be better to just have a `news: bool` parameter instead to avoid mixups.
Additionally, at least according to the api docs, `flags` cannot be provided on channel creation (this applies to all `clone` methods).